__tests__/runner.js

```diff
@@ -39,10 +42,12 @@ for (const testName of readdirSync(__dirname)) {
     });
     // If any diffs are detected, we'll assert the difference so that we get nice output.
     res.diffSet.forEach(function(entry) {
-      assert.strictEqual(
-        readFileSync(path.join(entry.path1, entry.name1), {encoding: 'utf8'}),
-        readFileSync(path.join(entry.path2, entry.name2), {encoding: 'utf8'}),
-      );
+      if (entry.type1 === 'file') {
```

Comment:
When @pika/web generates folders (`import httpVueLoader from '/web_modules/http-vue-loader/src/httpVueLoader.js'`), Node would throw an error trying to read a directory.

Reply:
Ah, you mean that if the two folders were found to have not matched, we'd get a folder object here instead? That makes sense, although it would be nice to still throw an assertion error if two directories don't match. 

Also since you're here, can we turn the `noDiffSet` option on for these tests? I was just re-reading the https://www.npmjs.com/package/dir-compare#use docs and it looks like it will speed up our tests.

Reply:
> can we turn the noDiffSet option on for these tests?

Oh nice! Didn’t know about that option. I think that would help.

But in general, I think we still need to have a check here because what’s breaking is `fs.readFileSync` trying to read a directory, so this should probably stay.

Reply:
Right, I think I'm imagining something like this:

```
if (entry.type1 === 'file') {
  // read files, and assert.
} else if (entry.type1 === 'directory') {
  // don't try to read files, just assert.
}
```

Although that assertion would have less info than the file assertion error message. I guess, are we guarenteed that a folder diff mismatch always contains a file diff mismatch? In that case, then I'd just rewrite to here to be a bit more explicit to readers:


```
if (entry.type1 !== 'file') {
  // NOTE: We only compare files so that we give the test runner a more detailed diff.
  return;
} 
// read files, and assert.
```

Reply:
Seems like the `noDiffSet` just disables the `diffSet` entirely (so we don’t get the nice output here as expected).

Enabling that would basically change the end of the test to:

```js
    var res = dircompare.compareSync(actualWebDependenciesLoc, expectedWebDependenciesLoc, {
      compareSize: true,
      noDiffSet: true,
    });
    return assert(res.same);
```

On my machine turning that off/on I couldn’t find any change in the time it took to run the tests (at least with the current number).

But it seems like the `diffSet`, no matter what, will always be an array of **every file it scanned** and not only the diffs (somewhat of a misnomer). On each item there’s a `state: 'equal'` that shows you 2 files/dirs are the same, and they’re in the `diffSet` only as proof they were scanned.

Anyway, I’m thinking we should probably leave everything as-is but just change the conditional to your last suggestion?

Reply:
Ah dang, okay thanks for looking into it. +1 for keeping as is